apps/src/templates/sectionProgressV2/SectionProgressV2.jsx

```diff
@@ -28,10 +30,26 @@ function SectionProgressV2({
     }
   }, [unitData, isLoadingProgress, isRefreshingProgress, scriptId, sectionId]);
 
+  React.useEffect(() => {
+    if (expandedLessonIds.length > 0) {
+      const hasValidatedLevel = unitData.lessons.some(
+        lesson =>
+          lesson.levels.some(level => level.isValidated) &&
+          expandedLessonIds.includes(lesson.id)
+      );
+      setIsViewingValidatedLevel(hasValidatedLevel);
+    } else {
+      setIsViewingValidatedLevel(false);
+    }
+  }, [expandedLessonIds, unitData]);
```

Comment:
Nice job using functional array methods here! I think `useMemo` is more appropriate for this case. Again, we're not causing any side effects such as calling an API, we're saving the results of a relatively expensive calculation. That's exactly what `useMemo` is for.
```suggestion
  const isViewingValidatedLevel = React.useMemo(() => {
    return unitData.lessons
      .filter(lesson => expandedLessonIds.includes(lesson.id))
      .some(lesson => lesson.levels.some(level => level.isValidated));
  }, [expandedLessonIds, unitData]);
```

Reply:
Thanks - this makes sense to me after reading the `useMemo` documentation.